src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/actions/RunSelectedTestAction.java

```diff
@@ -0,0 +1,253 @@
+      PsiClass psiClass = (PsiClass) element;
+      boolean isTestClass = false;
+      for (PsiMethod method : psiClass.getAllMethods()) {
+        isTestClass = isTestClass || isTestMethod(method);
+      }
+      if (isTestClass) {
+        presentation.setEnabled(true);
+        presentation.setVisible(true);
+      } else {
+        presentation.setEnabled(false);
+        presentation.setVisible(false);
+      }
+    }
+    else {
+      presentation.setEnabled(false);
+      presentation.setVisible(false);
+    }
+  }
+
+  /**
+   * Setup And Exectute a test configuration.
+   *
+   * @param event the event corresponding to an action.
+   * @param runInDebug whether or not the test should be run in debug.
+   */
+  void setupAndExecuteTestConfigurationFromEditor(AnActionEvent event, boolean runInDebug) {
+    PsiElement element = event.getData(LangDataKeys.PSI_ELEMENT);
+    PsiMethod method = null;
+    PsiClass psiClass = null;
+    if (element instanceof PsiMethod) {
+      method = (PsiMethod) element;
+      psiClass = method.getContainingClass();
+    }
+    if (element instanceof PsiClass) {
+      psiClass = (PsiClass) element;
+    }
+    final Project project = event.getProject();
+    if (project == null) {
+      return;
+    }
+    Editor editor = FileEditorManager.getInstance(project).getSelectedTextEditor();
+    if (editor == null) {
+      return;
+    }
+    final Document document = editor.getDocument();
+    VirtualFile virtualFile = FileDocumentManager.getInstance().getFile(document);
+
+    VirtualFile buckFile = BuckFileUtil.getBuckFile(virtualFile);
+    if (buckFile != null) {
+      String name = "";
+      String testSelectors = "";
+      if (method != null) {
+        name = psiClass.getName() + "#" + method.getName();
+        testSelectors = psiClass.getName() + "#" + method.getName();
+      } else {
+        name = element.getContainingFile().getName();
+        if (element instanceof PsiClass) {
+          testSelectors = psiClass.getQualifiedName();
+        }
+      }
+      createTestConfigurationFromContext(
+          name,
+          testSelectors,
+          project,
+          virtualFile,
+          buckFile,
+          runInDebug);
+    }
+  }
+
+  /**
+   * Setup And Exectute a test configuration from the project view.
+   *
+   * @param event the event corresponding to an action.
+   * @param runInDebug whether or not the test should be run in debug.
+   */
+  void setupAndExecuteTestConfigurationFromProject(AnActionEvent event, boolean runInDebug) {
+    PsiElement element = event.getData(LangDataKeys.PSI_ELEMENT);
+    PsiClass psiClass = null;
+    if (element instanceof PsiClass) {
+      psiClass = (PsiClass) element;
+    }
+    Project project = event.getData(PlatformDataKeys.PROJECT);
+    VirtualFile virtualFile = event.getData(PlatformDataKeys.VIRTUAL_FILE);
+    VirtualFile buckFile = BuckFileUtil.getBuckFile(virtualFile);
+    String name = "";
```

Comment:
You should try to use Java Optional instead of empty strings everywhere where there is an optional parameter.